app/main.py

```diff
@@ -1,3 +1,17 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    operations = {}
+
+    def wrapper(*args: Any) -> Any:
+        if args in operations:
+            print("Getting from cache")
+            return operations[args]
+        else:
+            print("Calculating new result")
+            result = func(*args)
+            operations[args] = result
+            return result
```

Comment:
You can use a single return statement out of the if/else statement instead of two inside it.